src/utils.ts

```diff
@@ -188,15 +485,73 @@ export namespace Utils {
     transaction: Transaction,
     accountId: string,
   ): boolean {
+    return verifyTxMultiSignedBy(transaction, accountId).length !== 0;
+  }
+
+  /**
+   *
+   * verifyTxMultiSignedBy checks if a transaction has been signed by one or more of
+   * the signers, returning a list of signers that were found to have signed the
+   * transaction.
+   *
+   * @function
+   * @memberof Utils
+   * @param {Transaction} transaction the signed transaction.
+   * @param {string[]} accountIDs The signers public keys.
+   * @example
+   * let keypair1 = Keypair.random();
+   * let keypair2 = Keypair.random();
+   * const account = new StellarSdk.Account(keypair1.publicKey(), "-1");
+   *
+   * const transaction = new TransactionBuilder(account, { fee: 100 })
+   *    .setTimeout(30)
+   *    .build();
+   *
+   * transaction.sign(keypair1, keypair2)
+   * Utils.verifyTxMultiSignedBy(transaction, [keypair1.publicKey(), keypair2.publicKey()])
+   * @returns {string[]} a list of signers that were found to have signed the transaction.
+   */
+  export function verifyTxMultiSignedBy(
+    transaction: Transaction,
+    ...accountIDs: string[]
+  ): string[] {
     const hashedSignatureBase = transaction.hash();
 
-    const keypair = Keypair.fromPublicKey(accountId);
+    const usedSignatures: boolean[] = new Array(transaction.signatures.length);
+    const signersFound = new Map<string, boolean>();
 
-    return !!transaction.signatures.find((sig) => {
-      return keypair.verify(hashedSignatureBase, sig.signature());
+    accountIDs.forEach((signer) => {
+      const keypair = Keypair.fromPublicKey(signer);
+
+      for (let i = 0; i < transaction.signatures.length; i++) {
+        if (usedSignatures[i]) {
```

Comment:
Given the following scenario

```
accountIDS = [account1, account2, account3,..., account10]
transaction.signatures = [account1, account2, account3]
```

By the time we process `account3` we know all signatures were consumed however we'll keep looping through account4 until 10th, and then loop 3 more times inside every each.

Reply:
Also if the signature is a txhash you could remove it as you loop through each signature

Reply:
Those were great suggestions, thanks @abuiles!

About removing the `txHash` signatures in the `verifyTxMultiSignedBy` metjod, I'm not sure if that's the right place for that, since this method is more generic and is supposed to return the list of public keys from accounts that have signed the given transaction...

Any thoughts, @leighmcculloch?

Reply:
>if the signature is a txhash

This is an optimization, so I don't feel strongly either way. Same with the optimizations to use a copy of the array.

These optimization might actually be slower because they involve allocating new memory (which is expensive), and deleting items in arrays (which is also expensive). These arrays are going to be very small, 1-4 in length, and in the rarest of cases at most only have 20 entries each.

Ultimately, I don't mind, but I think there is greater value in making this code as readable and simple as possible, and making the implementations of the different SDKs more similar will make it easier to maintain them, so I'd do that unless it contradicts the right way to do something in JavaScript.